godot-core/src/builtin/signal.rs

```diff
@@ -213,3 +216,310 @@ impl fmt::Display for Signal {
         write!(f, "{}", self.to_variant())
     }
 }
+
+// ----------------------------------------------------------------------------------------------------------------------------------------
+// Implementation of a rust future for Godot Signals
+#[cfg(since_api = "4.2")]
+mod futures {
+    use std::fmt::Display;
+    use std::future::Future;
```

Comment:
This could probably be extracted to its own file. The `#[cfg]` can then be placed on the import statement.